### PR TITLE
Create bills / lineitems async function

### PR DESCRIPTION
**Pull request by Shwetabhk**

### CODE REVIEW

**Comment by Shwetabhk** [comment]
12/30 are migration files, please ignore

File: apps/fyle/models.py [hunk 1]
```diff
@@ -76,36 +86,45 @@ class ExpenseGroup(models.Model):
     Expense Group
     """
     id = models.AutoField(primary_key=True)
+    fyle_group_id = models.CharField(max_length=255, unique=True, help_text='fyle expense group id report id, etc')
     workspace = models.ForeignKey(Workspace, on_delete=models.PROTECT,
                                   help_text='To which workspace this expense group belongs to')
     expenses = models.ManyToManyField(Expense, help_text="Expenses under this Expense Group")
-    description = models.CharField(max_length=255, help_text='Description')
+    description = JSONField(max_length=255, help_text='Description', null=True)
```
Comment (snarayanank2) [inline apps/fyle/models.py]: why is description a json field?

File: apps/quickbooks_online/tasks.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+from apps.fyle.models import ExpenseGroup
+from apps.tasks.models import TaskLog
+from apps.mappings.models import EmployeeMapping
+from apps.workspaces.models import QBOCredential
+
+from .models import Bill, BillLineitem
+from .utils import QBOConnector
+
+
+def create_bill(expense_group: ExpenseGroup, activity: TaskLog):
```
Comment (snarayanank2) [inline apps/quickbooks_online/tasks.py]: why do you need to pass the activity model? why dont you retrieve the right tasklog object based on task id.. I presume that is available somehow?

File: apps/quickbooks_online/views.py [hunk 3]
```diff
@@ -117,3 +128,48 @@ def get_departments(self, request, **kwargs):
...
+        Create bill from expense group
+        """
+        expense_group_ids = request.data.get('expense_group_ids', [])
+
+        assert_valid(expense_group_ids != [], 'Expense ids not found')
+
+        expense_groups = ExpenseGroup.objects.filter(
+            workspace_id=kwargs['workspace_id'], id__in=expense_group_ids).all()
+
+        activities = []
```
Comment (snarayanank2) [inline apps/quickbooks_online/views.py]: why name an object of type "TaskLog" as activity? that is confusing nomenclature. can we not call it task_logs or something like that?

File: apps/quickbooks_online/tasks.py (hunk 2 above)
Comment (snarayanank2) [reply apps/quickbooks_online/tasks.py]: This logic should be in a model class, not in tasks

**Pull request merged**